packages/x-data-grid/src/components/panel/filterPanel/GridFilterInputValue.tsx

```diff
@@ -62,17 +75,17 @@ function GridFilterInputValue(props: GridTypeFilterInputValueProps) {
 
   React.useEffect(() => {
     const itemPlusTag = item as ItemPlusTag;
-    if (itemPlusTag.fromInput !== id || item.value === undefined) {
-      setFilterValueState(String(item.value ?? ''));
+    if (itemPlusTag.fromInput !== id || item.value == null) {
```

Comment:
using `==` to also consider `null` values

Reply:
I'd rather explicitly use `x === null || x === undefined` rather than rely on the `==` operator, which I think we should not use in any situation.

Reply:
thanks for the review, but `x == null` is only loosely equal to `undefined`, no matter what it there in `item.value` if it is not `null` or `undefined` it always returns false? or  l'm missing something?